### PR TITLE
Feature/order entity

### PR DESCRIPTION
**Pull request by thebaker10**

### CODE REVIEW

File: docker/api/src/Domain/Order/User_order.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    #[Column(type: 'datetime', unique: false, nullable: false)]
+    private int $order_date;
+
+    #[Column(type: 'float', unique: false, nullable: false)]
+    private string $order_price;
+
+    #[ManyToOne(targetEntity: User::class)]
+    #[JoinColumn(name: 'user_id', referencedColumnName: 'user_id')]
+    private User $user;
+
```
Comment (noko12312) [inline docker/api/src/Domain/Order/User_order.php:36]:
```suggestion

    public function setUser(User $user): void{
        $this->user = $user;
    }
```

Comment (noko12312) [reply docker/api/src/Domain/Order/User_order.php:36]: needs function to set user

File: docker/api/src/Domain/Order/User_order.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+use Doctrine\ORM\Mapping\Entity;
+use Doctrine\ORM\Mapping\GeneratedValue;
+use Doctrine\ORM\Mapping\Id;
+use Doctrine\ORM\Mapping\JoinColumn;
+use Doctrine\ORM\Mapping\ManyToOne;
+use Doctrine\ORM\Mapping\Table;
+use JetBrains\PhpStorm\ArrayShape;
+use JsonSerializable;
+
+#[Entity, Table(name: 'user_order')]
```
Comment (noko12312) [inline docker/api/src/Domain/Order/User_order.php]:
```suggestion
#[Entity, Table(name: 'order')]
```

Comment (noko12312) [reply docker/api/src/Domain/Order/User_order.php]: It is more simple to just use order for table name. The class name User_order and file name User_order.php should probably be switched to just Order as well

File: docker/api/src/Application/Actions/Order/OrderPostAction.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    {
+
+        $payload = $this->request->getParsedBody();
+        $userId = $payload['userId'];
+        $orderDate = $payload['orderDate'];
+        $orderPrice = $payload['orderPrice'];
+        $order = new User_order($userId, $orderDate, $orderPrice);
+
+        $user = $this->em->getRepository(User::class)->find($userId);
+        $order->setProduct($user);
```
Comment (noko12312) [inline docker/api/src/Application/Actions/Order/OrderPostAction.php]:
```suggestion
        $order->setUser($user);
```

Comment (noko12312) [reply docker/api/src/Application/Actions/Order/OrderPostAction.php]: Calls setUser function instead of setProduct

**Review by noko12312 — CHANGES_REQUESTED** [review]

**Comment by thebaker10** [comment]
all changes requested added

**Review by noko12312 — APPROVED** [review]

**Pull request merged**